### DISCUSSION TITLE
Is the default learning rate a bit slow for music?

### DISCUSSION THREAD

**Discussion by Taikakim** [Stability-AI/stable-audio-tools#115] [General]
I have a dataset of somewhere around 150 of my own tracks I started training with using the default settings. Loss started from around 0.7 and even after eight hours of so training with a batch size of 38 on an A100 it only got down to around 0.66. This very different from my experience with image models where the loss with diffusion training generally hovers around 0.1 pretty much constantly.

Is it so that the original model just does not have that much music in (my style is kind of psychedelic trance/synth stuff), and there isn't just a lot to relate the training data with?

I'm now going to increase the LR by an order of magnitude just to see what happens. But it would also be interesting to see a full listing of the training data, or mainly, what kind of metadata was passed to see how I should rename my own files to place them close to positions where there is a lot of relevant data in the latent space. Or is that how this should work?

**Comment by Taikakim** [comment]
Okayyy, I made an unwrapped checkpoint at step 1200, batch size 38, and started a new train on that with a ten times higher learning rate of 5e-4, and the LR started climbing and the model to break down, so next maybe need to try something a bit more conservative:
![image](https://github.com/Stability-AI/stable-audio-tools/assets/21206467/11276615-e613-41c6-9dc3-27fce378f189)

**Comment by sskalnik** [comment]
A larger dataset size, such as 5000+ audio files, would probably improve the quality of the results.

This dataset consists of about 3500 audio files:
<img width="1998" height="926" alt="Screenshot 2025-10-07 231001" src="https://github.com/user-attachments/assets/ac903495-a73f-4393-b18a-1253e7c60ea5" />

Batch sizes were increased on each training run, from 2 to 4 to 8 to 16 to 32 to 40.

```
"optimizer_configs": {
            "diffusion": {
                "optimizer": {
                    "type": "AdamW8bit",
                    "config": {
                        "lr": 5e-5,
                        "betas": [0.9, 0.999],
                        "eps": 1e-8,
                        "weight_decay": 0.01,
                        "percentile_clipping": 100,
                        "block_wise": true
                    }
                },
                "scheduler": {
                    "type": "InverseLR",
                    "config": {
                        "inv_gamma": 1000000,
                        "power": 0.5,
                        "warmup": 0.995
                    }
                }
            }
        },
        ```
